src/ray/raylet/scheduling_queue.cc

```diff
@@ -56,6 +60,7 @@ std::vector<Task> SchedulingQueue::RemoveTasks(
   std::vector<Task> removed_tasks;
 
   // Try to find the tasks to remove from the waiting tasks.
+  removeTasksFromQueue(uncreated_actor_methods_, task_ids, removed_tasks);
```

Comment:
I'm a bit worried about `RemoveTasks` being expensive, but that's an issue for later.

Reply:
Yeah, I think eventually the plan is to make the implementation of `RemoveTasks` store a hashmap from `TaskID` to iterator so that we can remove it from the queue quickly.